packages/core/src/codewhisperer/activation.ts

```diff
@@ -337,7 +342,26 @@ export async function activate(context: ExtContext): Promise<void> {
             [...CodeWhispererConstants.securityScanLanguageIds],
             SecurityIssueCodeActionProvider.instance
         ),
-        vscode.commands.registerCommand('aws.amazonq.openEditorAtRange', openEditorAtRange)
+        vscode.commands.registerCommand('aws.amazonq.openEditorAtRange', openEditorAtRange),
+        auth.regionProfileManager.onDidChangeRegionProfile(() => {
+            // validate user's still has access to the selected customization
+            const selectedCustomization = getSelectedCustomization()
+            // no need to validate base customization which has empty arn
+            if (selectedCustomization.arn.length > 0) {
+                getAvailableCustomizationsList()
+                    .then((customizations) => {
+                        const r = customizations.find((it) => it.arn === selectedCustomization.arn)
+                        if (!r) {
+                            void switchToBaseCustomizationAndNotify().then()
+                        }
+                    })
+                    .catch((e) => {
+                        getLogger().error(
+                            `encounter error while validating selected customization on profile change: ${(e as Error).message}`
```

Comment:
```suggestion
                        getLogger().error(
                            `encounter error while validating selected customization on profile change: %s`, (e as Error).message
```

Reply:
https://github.com/aws/aws-toolkit-vscode/pull/7049/commits/35a5045629de765037c3751c2c386f885764fb54

Reply:
could you share what's the reason we prefer one over the other here